src/main/java/org/cactoos/iterable/Partitioned.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterable;
+
+import java.util.List;
+
+/**
+ * Iterable implementation for partitioning functionality.
+ *
+ * @author Sven Diedrichsen (sven.diedrichsen@gmail.com)
+ * @version $Id$
+ * @param <T> Partitions value type
+ * @since 0.29
+ */
+public class Partitioned<T> extends IterableEnvelope<List<T>> {
```

Comment:
@svendiedrichsen This class must be final. And why `IterableEnvelope<List<T>>`? Could not it be `IterableEnvelope<Iterable<T>>` or something like that?

Reply:
It could be `IterableEnvelope<Iterable<T>>` but this class is supposed to be a drop in replacement for Guava partitioning and I just have copied the interface.

Reply:
Made class final.

Reply:
This is supposed to be a drop in replacement for Guavas partitioning. Guava uses List as return value.